File: Data Structures & Algorithms/house-robber/submission-3.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        
        dp = []
        dp.append(nums[0])
        dp.append(max(nums[0],nums[1]))
        
        i = 2
        for num in nums[2:]:
            dp.append(max(num+dp[i-2], dp[i-1]))
            i+=1
        return dp[-1]

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {
       Set<Integer> set = new HashSet<>();

       for(int num:nums){
        set.add(num);
       }
       int counter=0, max=0;
       for(Integer num: set){
          if(!set.contains(num-1)){
            counter=1;
            int current=num;
            while(set.contains(current+1)){
              counter++;
              current++;
            }
            max = Math.max(counter, max);
          }
       }
       return max;
    }
}
